Mathlib/Topology/Inseparable.lean

```diff
@@ -70,38 +70,26 @@ infixl:300 " ⤳ " => Specializes
 
 /-- A collection of equivalent definitions of `x ⤳ y`. The public API is given by `iff` lemmas
 below. -/
-theorem specializes_TFAE ( x y : X ) :
+theorem specializes_TFAE (x y : X) :
     TFAE [x ⤳ y,
       pure x ≤ 𝓝 y,
       ∀ s : Set X , IsOpen s → y ∈ s → x ∈ s,
       ∀ s : Set X , IsClosed s → x ∈ s → y ∈ s,
       y ∈ closure ({ x } : Set X),
       closure ({ y } : Set X) ⊆ closure { x },
       ClusterPt y (pure x)] := by
-  -- todo: rewrite using `tfae_have` etc
-  apply_rules [tfae_of_cycle, Chain.cons, Chain.nil] <;> dsimp only [ilast']
-  · exact le_trans (pure_le_nhds _)
-  · exact fun h s hso hy => h (hso.mem_nhds hy)
-  · exact fun h s hsc hx => of_not_not fun hy => h (sᶜ) hsc.isOpen_compl hy hx
-  · exact fun h => h _ isClosed_closure (subset_closure rfl)
-  · exact fun h => closure_minimal (singleton_subset_iff.2 h) isClosed_closure
-  · rw [← principal_singleton, ← mem_closure_iff_clusterPt]
-    exact fun h => h (subset_closure rfl)
-  · refine fun h => (nhds_basis_opens _).ge_iff.2 fun U ⟨hyU, hUo⟩ => ?_
-    rw [← Ultrafilter.coe_pure, Ultrafilter.clusterPt_iff] at h
-    exact hUo.mem_nhds (h <| hUo.mem_nhds hyU)
-  -- tfae_have 1 → 2; exact pure_le_nhds _ . trans
-  -- tfae_have 2 → 3; exact fun h s hso hy => h hso . mem_nhds hy
-  -- tfae_have 3 → 4; exact fun h s hsc hx => of_not_not fun hy => h s ᶜ hsc . is_open_compl hy hx
-  -- tfae_have 4 → 5; exact fun h => h _ isClosed_closure subset_closure <| mem_singleton _
-  -- tfae_have 6 ↔ 5; exact is_closed_closure.closure_subset_iff.trans singleton_subset_iff
-  -- tfae_have 5 ↔ 7; rw [ mem_closure_iff_clusterPt, principal_singleton ]
-  -- tfae_have 5 → 1
-  -- · refine' fun h => nhds_basis_opens _ . ge_iff . 2 _
-  --   rintro s ⟨ hy , ho ⟩
-  --   rcases mem_closure_iff . 1 h s ho hy with ⟨ z , hxs , rfl : z = x ⟩
-  --   exact ho.mem_nhds hxs
-  -- tfae_finish
+  tfae_have 1 → 2; exact (pure_le_nhds _).trans
+  tfae_have 2 → 3; exact fun h s hso hy => h (hso.mem_nhds hy)
+  tfae_have 3 → 4; exact fun h s hsc hx => of_not_not fun hy => h (sᶜ) hsc.isOpen_compl hy hx
+  tfae_have 4 → 5; exact fun h => h _ isClosed_closure (subset_closure <| mem_singleton _)
+  tfae_have 6 ↔ 5; exact isClosed_closure.closure_subset_iff.trans singleton_subset_iff
+  tfae_have 5 ↔ 7; · rw [mem_closure_iff_clusterPt, principal_singleton]
```

Comment:
```suggestion
  tfae_have 5 ↔ 7; rw [mem_closure_iff_clusterPt, principal_singleton]
```

Reply:
Maybe I don't understand why you wrote it this way. It seems strange to focus for the `rw` proofs but not for the `exact` proofs?

Reply:
`exact` always closes the goal, `rw` doesn't necessarily. I think it is better with the `· rw`

Reply:
This was copied straight from mathport, but I regularized the formatting a bit.